cypress/integration/select-controller.spec.js

```diff
@@ -3,77 +3,198 @@ context('SelectController', () => {
-    it('selects an item from the list', () => {
-      cy.get('.custom-select').click()
-      cy.get('.dropdown-items div:nth-child(2)').click()
-      cy.get('.custom-select').should('have.value', 'Two')
-      cy.get('select').should('have.value', '2')
+    context('using the mouse', () => {
+      it('selects an item from the list', () => {
+        cy.get('#first-select .select').children('.custom-select').click()
+        cy.get('.dropdown-items div:nth-child(2)').click()
+        cy.get('#first-select .select')
+          .children('.custom-select')
+          .should('have.value', 'Two')
+        cy.get('select').should('have.value', '2')
+      })
+    })
+
+    context('using the keyboard', () => {
+      it('selects an item from the list', () => {
+        cy.get('#first-select .select').children('.custom-select').click()
+        cy.get('#first-select .dropdown-container .search-input')
+          .type('{DOWNARROW}')
+          .type('{ENTER}')
+        cy.get('#first-select .select')
+          .children('.custom-select')
+          .should('have.value', 'One')
+        cy.get('select').should('have.value', '1')
+      })
+
+      it('selects an item after going down and up', () => {
+        cy.get('#first-select .select').children('.custom-select').click()
+        cy.get('#first-select .dropdown-container .search-input').type(
+          '{DOWNARROW}{DOWNARROW}{DOWNARROW}{DOWNARROW}{DOWNARROW}'
+        )
+
+        cy.get('.dropdown-items div:nth-child(5)').should(
+          'have.class',
+          'selected'
+        )
+
+        cy.get('#first-select .dropdown-container .search-input')
+          .type('{UPARROW}')
+          .type('{ENTER}')
+
+        cy.get('#first-select .select')
+          .children('.custom-select')
+          .should('have.value', 'Four')
+        cy.get('select').should('have.value', '4')
+      })
     })
   })
 
-  context('using the keyboard', () => {
-    it('selects an item from the list', () => {
-      cy.get('.custom-select').click()
-      cy.get('.search-input')
-        .type('{DOWNARROW}')
-        .type('{ENTER}')
-      cy.get('.custom-select').should('have.value', 'One')
-      cy.get('select').should('have.value', '1')
+  context('select with display text, values and data info', () => {
+    it('displays the selected item', () => {
+      cy.get('#second-select .select')
+        .children('.custom-select')
+        .should('have.value', 'Custom Value')
     })
 
-    it('selects an item after going down and up', () => {
-      cy.get('.custom-select').click()
-      cy.get('.search-input').type(
-        '{DOWNARROW}{DOWNARROW}{DOWNARROW}{DOWNARROW}{DOWNARROW}'
+    it('displays the list of available options', () => {
+      cy.get('#second-select .select').children('.custom-select').click()
+      cy.get('.dropdown-items div').should('have.length', 5)
+      cy.get('.dropdown-items div:nth-child(1)').should(
+        'contains.text',
+        'Text on right'
```

Comment:
Si te funciona contains con 's' al final? no deberia ser `.should('contain.text', 'Text on right')`